Add CompleteCommand

jia1 opened the pull request.

jia1 committed:
Create CompleteCommand (still in progress)

Task still gets updated to incomplete. Need to debug more.

jia1 committed:
Remove a completed TODO

jia1 committed:
Fix no update bug from previous commit

EditCommand keeps creating a new Task and intercepting the status at
initMembers will update the status for real.

jia1 committed:
Show invalid message when invalid Complete command

Imitate DeleteCommand

jia1 committed:
Merge branch 'master' into new/command/complete

alexfjw commented on src/main/java/seedu/taskman/logic/commands/EditCommand.java:
```diff
@@ -18,9 +18,9 @@
-    // UG/DG
+    // KIV: Let parameters be objects. We can easily generate the usage in that case.
+    // TODO: Update message. I'm so confused about the order of the arguments urgh...
+    // UG/DG (what is this for?)
```
for the day we make the changes

jia1 replied:
?

alexfjw replied:
Just remove the 3 lines. Might not make much sense anymore, given our current direction.

alexfjw commented on src/main/java/seedu/taskman/logic/commands/CompleteCommand.java:
```diff
@@ -0,0 +1,25 @@
+    public static final String MESSAGE_SUCCESS = "Task completed: %1$s";
+
+    public CompleteCommand(int targetIndex,
+            @Nullable String title, @Nullable String deadline, @Nullable String status,
```
How about using a constructor which takes only the targetIndex ?

alexfjw commented on src/main/java/seedu/taskman/logic/parser/CommandParser.java:
```diff
@@ -209,6 +209,25 @@ private Command prepareEdit(String args) {
+
+        return new CompleteCommand(
+		        index.get(),
+		        null,
```
As above, constructor which only takes targetIndex would be more appropriate. Don't need to put the nulls.

alexfjw commented on src/main/java/seedu/taskman/logic/commands/EditCommand.java:
```diff
@@ -98,35 +98,39 @@ private void initMembers(ArgumentContainer argsContainer) throws IllegalValueExc
+            	);
+            	task.status = argsContainer.status == null
+                        ? beforeEdit.getStatus().orElse(null)
+                        : new Status(argsContainer.status);
```
Its supposed to edit the task to completed here. (Not check the args passed in to the complete command)

alexfjw replied:
Or maybe it should take in arguments for the 'status' field. Depends on what you have in mind for complete. 

We could either go with the twin commands complete & incomplete, 
(revise the naming as you see fit)

or
Have complete take in different arguments to support setting to completed & uncompleted.

jia1 replied:
This one can be easily changed. KIV.

jia1 committed:
Replace PERSON with EVENT

jia1 committed:
Remove inheritance and reduce number of params

1. CompleteCommand no longer extends EditCommand
2. CompleteCommand constructor has only 1 parameter
3. execute method deeply resembles DeleteCommand and EditCommand
4. CommandParser is updated accordingly

jia1 committed:
Clean up for Edit and Complete commands

1. Remove the three random comments
2. Add a note for code that was added but later not needed in EditCommand
(this code is commented out)
3. Add comments for CompleteCommand

jia1 committed:
Add tests for CompleteCommand

Similar to tests for DeleteCommand

jia1 commented:
OKAY CompleteCommand takes 1 arg, works, and has tests!!!! @CS2103AUG2016-W10-C1/developers
